wrangler-service/src/main/java/io/cdap/wrangler/service/directive/CommonExecutor.java

```diff
@@ -0,0 +1,128 @@
+ */
+public class CommonExecutor {
+
+  private final Function<TransientStore, ExecutorContext> contextProvider;
+  private final DirectiveRegistry composite;
+
+  CommonExecutor(Function<TransientStore, ExecutorContext> contextProvider, DirectiveRegistry composite) {
+    this.contextProvider = contextProvider;
+    this.composite = composite;
+  }
+
+  /**
+   * Execute directives on the provided data sample and return the results
+   * @param namespace String namespace
+   * @param directives {@link List<String>} of directives to be applied
+   * @param sample {@link List<Row>} of sample data
+   * @return {@link List<Row>} result after applying the directives
+   * @throws DirectiveParseException if there is an issue
+   */
+  public List<Row> executeDirectives(String namespace, List<String> directives,
+                                     List<Row> sample) throws DirectiveParseException {
+    RecipePipelineExecutor executor = new RecipePipelineExecutor();
+    if (!directives.isEmpty()) {
+      GrammarMigrator migrator = new MigrateToV2(directives);
+      String migrate = migrator.migrate();
+      RecipeParser recipe = new GrammarBasedParser(namespace, migrate, composite);
+      recipe.initialize(new ConfigDirectiveContext(DirectiveConfig.EMPTY));
+      try {
+        executor.initialize(recipe, contextProvider.apply(new DefaultTransientStore()));
+        sample = executor.execute(sample);
+      } catch (RecipeException e) {
+        throw new BadRequestException(e.getMessage(), e);
+      }
+
+      List<ErrorRecordBase> errors = executor.errors()
+        .stream()
+        .filter(ErrorRecordBase::isShownInWrangler)
+        .collect(Collectors.toList());
+      if (errors.size() > 0) {
+        throw new ErrorRecordsException(errors);
+      }
+
+      executor.destroy();
+    }
+    return sample;
+  }
+
+  /**
+   * Utility method for converting {@link List} of {@link Row} to byte array
+   * @param rows {@link List} of {@link Row}
+   * @return byte[] equivalent for the rows
+   * @throws IOException if an I/O error occurs
+   */
+  public static byte[] convertToBytes(List<Row> rows) throws IOException {
+    try (ByteArrayOutputStream bos = new ByteArrayOutputStream();
+         ObjectOutput out = new ObjectOutputStream(bos)) {
+      out.writeObject(rows);
+      out.flush();
+      return bos.toByteArray();
+    }
+  }
+
+  /**
+   * Utility method for converting byte[] to {@link List} of {@link Row}
+   * @param data byte[] with data
+   * @return {@link List} of {@link Row}
+   * @throws IOException if an I/O error occurs
+   * @throws ClassNotFoundException if the class being read as is not found
+   */
+  public static List<Row> convertToRows(byte[] data) throws IOException, ClassNotFoundException {
+    List<Row> rows = new ArrayList<>();
+    if (data != null) {
+      try (ObjectInputStream ois = new ObjectInputStream(new ByteArrayInputStream(data))) {
+        rows = (List<Row>) ois.readObject();
```

Comment:
would this always be object of list of rows? Trying to see what is data in case of error is returned from `runTask`

Reply:
Yes, if there is an error an exception will be thrown but wrong data will not be returned. Reusing ObjectSerDe instead of this method now.